src/SwipeButton.tsx

```diff
@@ -63,140 +77,177 @@ export type SwipeButtonCommonProps = {
+  underlayStyle,
+  disabled,
+  Icon,
+  iconContainerStyle,
+  containerStyle,
+  onComplete,
+  onSwipeEnd = () => {},
+  onSwipeStart = () => {},
 }: SwipeButtonProps) => {
-
-    const [endReached, setEndReached] = useState<boolean>(false);
-    const opacity = disabled ? 0.5 : 1;
-    const opacityStyle = { opacity };
-    const [translateX] = useState<Animated.Value>(new Animated.Value(0));
-    const scrollDistance = width - (completeThresholdPercentage / 100) - height;
-    const completeThreshold = scrollDistance * (completeThresholdPercentage / 100);
-
-    const animateToStart = () => {
-        Animated.spring(translateX, { toValue: 0, tension: 10, friction: 5, useNativeDriver: false }).start();
-
-        return setEndReached(false);
-    };
-
-    const animateToEnd = () => {
-        onComplete();
-        Animated.spring(translateX, { toValue: scrollDistance, tension: 10, friction: 5, useNativeDriver: false }).start();
-
-        return setEndReached(true);
-    };
-
-    const onMove = (_: GestureResponderEvent, gestureState: PanResponderGestureState) => {
-        if (disabled) {
-            return false;
-        }
-
-        if (gestureState.dx < 0 || gestureState.dx > scrollDistance) {
-            return Animated.event([{ dx: translateX }], { useNativeDriver: false })({
-                ...gestureState,
-                dx: gestureState.dx < 0 ? 0 : scrollDistance,
-            });
-        }
-        
-        return Animated.event([{ dx: translateX }], { useNativeDriver: false })(gestureState);
-    };
-
-    const onRelease = () => {
-        if (disabled) {
-            return;
-        }
-
-        if (endReached) {
-            return animateToStart();
-        }
-
-        const isCompleted = translateX._value! >= completeThreshold;
-        
-        return isCompleted
-            ? animateToEnd()
-            : animateToStart();
-    };
-
-    const panResponser = () => PanResponder.create({
-        onPanResponderGrant: onSwipeStart,
-        onPanResponderEnd: onSwipeEnd,
-        onStartShouldSetPanResponder: () => false,
-        onMoveShouldSetPanResponder: () => true,
-        onMoveShouldSetPanResponderCapture: () => false,
-        onPanResponderMove: onMove,
-        onPanResponderRelease: onRelease,
+  const [dimensions, setDimensions] = useState({ width: 0, height: 0 });
+  const [endReached, setEndReached] = useState<boolean>(false);
+  const opacity = disabled ? 0.5 : 1;
+  const [translateX] = useState<Animated.Value>(new Animated.Value(0));
+  const scrollDistance =
+    dimensions.width - completeThresholdPercentage / 100 - dimensions.height;
+  const completeThreshold =
+    scrollDistance * (completeThresholdPercentage / 100);
+
+  const onLayoutContainer = async (e: LayoutChangeEvent) => {
+    const { width, height } = e.nativeEvent.layout;
+    setDimensions({ width, height });
+  };
+
+  const animateToStart = () => {
+    Animated.spring(translateX, {
+      toValue: 0,
```

Comment:
No that you are revamping this, we could add the spring configuration as a prop. Let me know what you think about doing this

Reply:
Hmm interesting, I think it would be a good idea. I personally don't have a need for this now but I agree in the philosophy that the more configurable it is for the end user, the better.

Reply:
You do not have to do this. It was just an idea for additional changes. I will instead create an #2 for implementing this. I might get to it my self